src/features/world/scenes/BeachScene.ts

```diff
@@ -361,6 +326,45 @@ export class BeachScene extends BaseScene {
       }
     });
 
+    const piratePotionEquipped = [
+      ...Object.values(this.gameState.bumpkin?.equipped ?? {}),
+      ...Object.values(this.gameState.farmHands.bumpkins).flatMap((bumpkin) =>
+        Object.values(bumpkin.equipped),
+      ),
+    ].includes("Pirate Potion");
+    const openedAt = this.gameState.pumpkinPlaza.pirateChest?.openedAt ?? 0;
+    const hasOpened =
+      !!openedAt &&
+      new Date(openedAt).toISOString().substring(0, 10) ===
+        new Date().toISOString().substring(0, 10);
+
+    if (hasFeatureAccess(this.gameState, "PIRATE_CHEST")) {
+      if (piratePotionEquipped && !hasOpened) {
```

Comment:
What would happen if a player goes to the chest and then equips a potion?

To make this more dynamic, I would suggest setting up a listener that will react and render whenever the equipped changes.

You can set up a listener as such:

```
const listener = (e: EventObject) => {
        if (e.type === "bumpkin.equipped") {
          this.updatePirateChest(); // Some function you would put the render logic in
        }
      };

      this.gameService.onEvent(listener);

      this.events.on("shutdown", () => {
        this.gameService.off(listener);
      });
```

This ensures that when the potion is equipped, the lock is removed